dev-tools/omdb/src/bin/omdb/nexus.rs

```diff
@@ -1548,3 +1570,140 @@ async fn cmd_nexus_sled_expunge(
+        .await
+        .context("loading latest collection")?
+    {
+        Some(collection) => {
+            let disk_identity = omicron_common::disk::DiskIdentity {
+                vendor: physical_disk.vendor.clone(),
+                serial: physical_disk.serial.clone(),
+                model: physical_disk.model.clone(),
+            };
+
+            let mut sleds_containing_disk = vec![];
+
+            for (sled_id, sled_agent) in collection.sled_agents {
+                for sled_disk in sled_agent.disks {
+                    if sled_disk.identity == disk_identity {
+                        sleds_containing_disk.push(sled_id);
+                    }
+                }
+            }
+
+            match sleds_containing_disk.len() {
+                0 => {}
+                1 => {
+                    eprintln!(
+                        "WARNING: physical disk {} is PRESENT in the most \
+                         recent inventory collection (spotted at {}). It is \
+                         dangerous to expunge a disk that is still running, and \
+                         safer to expunge a disk from a system where it has been \
+                         removed. Are you sure you want to proceed anyway?",
+                        args.physical_disk_id, collection.time_done,
+                    );
+                    let confirm = read_with_prompt("y/N")?;
+                    if confirm != "y" {
+                        eprintln!("expungement not confirmed: aborting");
+                        return Ok(());
+                    }
+                }
+                _ => {
+                    // This should be impossible due to a unique database index,
+                    // "vendor_serial_model_unique".
+                    //
+                    // Even if someone tried moving a disk, it would need to be
+                    // decommissioned before being re-commissioned elsewhere.
+                    //
+                    // However, we still print out an error message here in the
+                    // (unlikely) even that it happens anyway.
+                    eprintln!(
+                        "ERROR: physical disk {} is PRESENT MULTIPLE TIMES in \
+                        the most recent inventory collection (spotted at {}).
+                        This should not be possible, and is an indication of a \
+                        database issue.",
+                        args.physical_disk_id, collection.time_done,
+                    );
+                    bail!("Physical Disk appeared on multiple sleds");
+                }
+            }
+        }
+        None => {
+            eprintln!(
+                "ERROR: cannot verify that the physical disk inventory status \
+                 because there are no inventory collections present. Please \
+                 make sure that the physical disk has been physically removed, \
+                 or ensure that inventory may be collected."
```

Comment:
Should we still allow continuing (with a prompt) here? If not, should we remove the suggestion to remove the disk and only say "Please ensure inventory may be collected"?

Reply:
I'm still leaning towards "bail early if we can't read the inventory" -- because that's a signal something is *really* wrong -- but I've updated the error message to be more appropriate (advise about inventory specifically).